connector/ws/src/lib.rs

```diff
@@ -82,8 +90,30 @@ impl<S: connector::Supervisor> Inner<S> {
             .install_default()
             .unwrap();
 
-        let token_ser_string =
-            serde_json::to_string(&self.config.token).expect("failed to re-serialize token");
+        let token = match self.config.token {
+            WsConnectorConfigToken::TokenFile { ref token_file } => {
+                let token_base64 = tokio::fs::read(&token_file).await.map_err(|io_err| {
+                    WsConnectorError::Config(ConfigError::IoError(token_file.clone(), io_err))
+                })?;
+
+                let token = base64::prelude::BASE64_STANDARD
+                    .decode(&token_base64)
+                    .map_err(|_| ())
+                    .and_then(|token_bytes| Ok(AuthToken(token_bytes.try_into().map_err(|_| ())?)))
+                    .map_err(|_| {
+                        WsConnectorError::Config(ConfigError::InvalidToken(
+                            "Provided token is not a valid base64-encoded string of the required length"
```

Comment:
This should probably just be the `#[error()]` string.

Reply:
Okay, what would I put in this string then?

Reply:
I mean, I can convert both of these error variants to strings and pass that through?

Reply:
Was thinking of just making `InvalidToken` a unit variant, e.g.:
```rs
enum ConfigError {
  // [snip]
  #[error("Provided token is not a valid base64-encoded string of the required length")]
  InvalidToken,
}
```